pkg/trace/api/debugger.go

```diff
@@ -23,34 +23,48 @@ const (
 	// logsIntakeURLTemplate specifies the template for obtaining the intake URL along with the site.
 	logsIntakeURLTemplate = "https://http-intake.logs.%s/api/v2/logs"
 
-	// logsIntakeMaximumTagsLength is the maximum number of characters we send as ddtags.
-	logsIntakeMaximumTagsLength = 4001
+	debuggerDiagnosticsURLTemplate = "https://debugger-intake.%s/api/v2/debugger"
+
+	// intakeMaximumTagsLength is the maximum number of characters we send as ddtags.
```

Comment:
Can you clarify where this is coming from? Is this is an intake-imposed limit?

Reply:
We had a bug which made the tags accumulate in memory and we added the limit to prevent this growing indefinitely in the future: https://github.com/DataDog/datadog-agent/commit/543fbb575f1dcb1efb06fe1effcc89a4d3fabc14

@ojung tested the bugfix and set the value, I didn't find anything in the intake docs about a limit for query string ddtags. I made a simple test now and managed to upload an event with 39999 chars in the query string for ddtags, so the limit is much higher than this.

I wouldn't change this value now though, we shouldn't hit it since we're only adding container and infrastructure tags in the proxy forwarder.

Reply:
The query string tags are probably included in the total event size (haven't tested this assumption) - in that case it make sense to keep a lower limit.

Reply:
Tags customers add as `DD_TAGS` are also included and a couple other tags from the tracer library. Our debugger snapshot uploads can become quite large, so we want to have a conservative limit on the tags added to avoid hitting the max request size limit of EVP intake. I'm pulling this from memory now, but I think I verified it back then.

Reply:
Gotcha. Any context you can provide here (without leaking internal data) will be helpful. 4,001 is just a very specific number and without any context someone could definitely be confused by how that number was chosen later down the line. I'm still not totally clear why it's not just 4000, for example.

Reply:
I've added a clarifying comment on the variable saying that the limit doesn't come from the event platform. It's a safeguard we've added so that the tags don't grow unbounded and they don't take up too much from the event payload we need for the data we capture.